### PR TITLE
Added /billing/gift endpoint

### PR DESCRIPTION
**Pull request by liam-lloyd**
In service of our forthcoming bulk gifting feature, we need an endpoint that can take in many emails and efficiently issue a gift to each of them. Such an endpoint (which can also handle gifts to individual users) is added in this commit.

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -345,3 +347,31 @@ Queues thumbnail generation tasks for all non-root folders created between `begi
   bannerImage: string (URL)
 }
 ```
+
+## Billing
+
+### POST /gift
```
Comment (cecilia-donnelly) [inline API.md:353]: It looks like this is `billing/gift` -- even though this is in the "Billing" section I think above we put the whole path for things like `archive/featured`. Can you use `billing/gift` here to be consistent?

File: src/email/service.ts [hunk 2]
```diff
@@ -97,3 +97,42 @@ export const sendArchiveStewardNotification = async (
...
+  const fullNameResult = await db.sql<{ fullName: string }>(
+    "email.queries.get_full_name_by_account_email",
+    { email: fromEmail }
+  );
+  if (fullNameResult.rows[0] === undefined) {
+    throw new Error(`Account with primary email ${fromEmail} not found`);
+  }
+  const { fullName } = fullNameResult.rows[0];
+  await sendEmail(
+    "invitation-from-relationship",
```
Comment (cecilia-donnelly) [inline src/email/service.ts:117]: For some reason the email I got locally looks different than the one I got from the backend invitation.

stela:
Screenshot 2023-10-27 at 15-52-19 Emptied database again is gifting you $10 of storage - cecilia@permanent org - The Permanent Legacy Foundation Mail


back-end:
Screenshot 2023-10-27 at 15-52-35 remodel is gifting you $10 of storage - cecilia@permanent org - The Permanent Legacy Foundation Mail


But this looks like the right template name, so I'm not sure what's going on.

Comment (cecilia-donnelly) [reply src/email/service.ts:117]: Oh duh it looks like my PR (https://bitbucket.org/permanent-org/email/pull-requests/53) broke the template so the email is only sending the first part of the text...I'll fix that.

**Review by cecilia-donnelly — APPROVED** [review]
A couple small requests but this looks good! Thanks @liam-lloyd . And wow, TDD yields lots of tests. :)

File: src/email/service.ts (hunk 2 above)
Comment (cecilia-donnelly) [reply src/email/service.ts:117]: Fix: https://bitbucket.org/permanent-org/email/pull-requests/55

File: src/fixtures/create_test_invites.sql [hunk 3]
```diff
@@ -1,4 +1,5 @@
 INSERT INTO
   invite (email, byAccountId, token, status, type)
 VALUES
-  ('test@permanent.org', 2, 'earlyb1rd', 'status.invite.revoked', 'type.invite.invite_code');
+  ('test@permanent.org', 2, 'earlyb1rd', 'status.invite.revoked', 'type.invite.invite_code'),
+  ('test+already_invited@permanent.org', 3, 'earlyb1rd', 'status.invite.revoke.pending', 'type.invite.invite_early_access');
```
[the remark this answers is not in the document]
Comment (liam-lloyd) [reply src/fixtures/create_test_invites.sql:5]: Oh, I just meant for this to be `status.invite.pending`; I'll fix that.

**Pull request merged**